pennylane/transforms/qcut.py

```diff
@@ -210,3 +213,235 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
             order += 1
 
     return graph
+
+
+@dataclass(frozen=True)
+class CutConfig:
+    """
+    Container wrapping partitioning parameters for passing to the automatic cutter.
+    """
+
+    k: int
+    imbalance: float
+    fragment_wires: List[int] = None
+    fragment_gates: List[int] = None
```

Comment:
Exactly, kahypar has the functionality to optionally pass a list of max fragment sizes if there's such a constraint. However since partition nodes corresponds to gates, it's only applicable to `fragment_gates` rather than `fragment_wires`. For `fragment_wires` to achieve similar functionality, I don't have a good idea right now other than sampling a bunch of partition trials and pick the one that satisfies the condition, if there's any. Or, we could remove `fragment_wires` for now, though it's realistically the more useful one.